airbyte-integrations/bases/connector-acceptance-test/Dockerfile

```diff
@@ -1,39 +1,40 @@
-FROM python:3.9.11-alpine3.15 as base
+FROM python:3.10.12 as base
 
 # build and load all requirements
 FROM base as builder
 WORKDIR /airbyte/integration_code
 
 # upgrade pip to the latest version
-RUN apk --no-cache upgrade \
+RUN apt-get update \
     && pip install --upgrade pip \
-    && apk --no-cache add tzdata build-base
+    && apt-get install tzdata
 
-COPY setup.py ./
-# install necessary packages to a temporary folder
-RUN python setup.py egg_info
-RUN pip install --prefix=/install -r *.egg-info/requires.txt
+COPY requirements.txt ./
+RUN pip install --prefix=/install -r requirements.txt
 
 # build a clean environment
 FROM base
+ENV ACCEPTANCE_TEST_DOCKER_CONTAINER 1
+
 WORKDIR /airbyte/connector_acceptance_test
+RUN echo "Etc/UTC" > /etc/timezone
+# Bash is installed for more convenient debugging.
+RUN apt-get install bash curl
+# Docker is installed to make it possible to run dagger
+ENV DOCKER_VERSION = "24.0.2"
+RUN curl -fsSL https://get.docker.com | sh
 
 # copy all loaded and built libraries to a pure basic image
 COPY --from=builder /install /usr/local
 # add default timezone settings
 COPY --from=builder /usr/share/zoneinfo/Etc/UTC /etc/localtime
-RUN echo "Etc/UTC" > /etc/timezone
-# Bash is installed for more convenient debugging.
-RUN apk --no-cache add bash
-
-ENV ACCEPTANCE_TEST_DOCKER_CONTAINER 1
 
 # copy payload code only
-COPY pytest.ini setup.py ./
+COPY pyproject.toml ./
```

Comment:
Two things we can improve here
1. Using poetry to take advantage of the lock file (deterministic builds, development dependencies match production deps, etc)
2. Install before we load the rest of the code (In docker we we copy just the pyproject.toml and poetry.lock and run `poetry install` before copying the rest of the code then our dependencies are cached and our builds are faster


```dockerfile
RUN pip install poetry==1.5.1
RUN poetry config virtualenvs.create false
COPY poetry.lock pyproject.toml ./
RUN poetry install --no-dev --no-interaction --no-ansi
COPY connector_acceptance_test ./connector_acceptance_test
```

https://stackoverflow.com/questions/53835198/integrating-python-poetry-with-docker
https://python-poetry.org/docs/#ci-recommendations